src/main/java/com/intellij/AbstractNotNullInstrumenterTask.java

```diff
@@ -56,6 +56,10 @@
 
     @Parameter
     private boolean implicit;
+
+    @Parameter
+    private boolean warnInsteadOfThrow;
```

Comment:
I'm thinking this may make more sense as a "mode" because this flag would be useless if `implicit` is false, right?.
Of course we don't want to break the maven api (or do you?). Just thinking.
It might make sense to allow `<mode>implicit</mode>`, `<mode>warn</mode>`, (default) `<mode>explicit</mode>` and remove the `<implicit>` tag.

Reply:
I was also thinking a little bit about modes, but then more like "throw" would be a mode and "warn" another, but since we don't have a third mode I felt like keeping it simple.

The warnInsteadOfThrow could be relevant and usable for both of explicit and implicit modes I think. Both explicit and implicit mode can throw. So both should be able to change those throws to warn.